pkg/pool/pool.go

```diff
@@ -85,6 +91,69 @@ func NewResourcePool(log logr.Logger, poolConfig *config.WarmPoolConfig, usedRes
 	return pool
 }
 
+// ReSync syncs state of upstream with the local pool. If local resources have additional
+// resource which doesn't reflect in upstream list then these resources are removed. If the
+// upstream has additional resources which are not present locally, these resources are added
+// to the warm pool. During ReSync all Create/Delete operations on the Pool should be halted
+// but Assign/Free on the Pool can be allowed.
+func (p *pool) ReSync(upstreamResource []string) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	// This is possible if two Re-Syn were requested at same time
```

Comment:
I don't think it's possible, but still wanted to add a safeguard for now.

Reply:
> but still wanted to add a safeguard for now.

Once you feel confident on other parts of the code, you could remove this. Otherwise, it can cause confusion to the reader further down the line.